### ISSUE TITLE
Dynamic import path does not work

### ISSUE THREAD

**Issue by WayneUong** [meteor/meteor#8744]
This works: `import('/imports/client/main.js');`
This doesn't work: `import('/imports/client/' + 'main.js');`

**Comment by dr-dimitru** [comment]
@WayneUong does this produce an error? error stack?

**Comment by WayneUong** [comment]
I got this error:

```
modules-runtime.js?hash=8587d18…:231 Uncaught (in promise) Error: Cannot find module '/imports/client/main.js'
    at makeMissingError (modules-runtime.js?hash=8587d18…:231)
    at Function.require.resolve (modules-runtime.js?hash=8587d18…:263)
    at Module.resolve (modules-runtime.js?hash=8587d18…:121)
    at modules-runtime.js?hash=8587d18…:183
    at meteor.js?hash=27829e9…:1105
    at <anonymous>
```

**Comment by benjamn** [comment]
This is a deliberate design decision and not a bug: https://github.com/meteor/meteor/pull/8327#issuecomment-280876264

**tl;dr** If you indicate elsewhere in your client code that the module should be made available (using `import(id)` where `id` is a statically analyzable string literal), then any dynamic expression will work in other places. However, Meteor will not make modules available for import unless your code expresses the dependency at build time.

**Comment by dr-dimitru** [comment]
@benjamn it's good to know, we should explicitly mention this in the [documentation](https://guide.meteor.com/structure.html#intro-to-import-export)

**Comment by benjamn** [comment]
@dr-dimitru good point!

**Comment by hwillson** [comment]
Thanks @dr-dimitru - I've created https://github.com/meteor/guide/issues/641 so we don't lose track of this.
